src/Microsoft.Framework.PackageManager/SourceControl/GitSourceControlProvider.cs

```diff
@@ -0,0 +1,177 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.Framework.PackageManager.SourceControl
+{
+    internal class GitSourceControlProvider: SourceControlProvider
+    {
+        private const string RepoApp = "git";
+        
+        private const string RepoUrlKey = "url";
+        private const string CommitHashKey = "commit";
+        private const string ProjectPathKey = "path";
+
+        private bool? _isInstalled;
+
+        public GitSourceControlProvider(Reports buildReports)
+            :base(buildReports)
+        {
+        }
+
+        public override bool IsInstalled
+        {
+            get
+            {
+                if (!_isInstalled.HasValue)
+                {
+                    _isInstalled = ProcessUtilities.ExecutableExists(RepoApp);
+                }
+
+                return _isInstalled.Value;
+            }
+        }
+
+        public override void AddMissingSnapshotInformation(string folderName, IDictionary<string, string> snapshotInformation)
+        {
+            if (!snapshotInformation.ContainsKey(RepoUrlKey) || string.IsNullOrEmpty(snapshotInformation[RepoUrlKey]))
+            {
+                throw new ArgumentNullException("The repository URL must be specified.");
+            }
+
+            if (!snapshotInformation.ContainsKey(CommitHashKey))
+            {
+                snapshotInformation[CommitHashKey] = GetHEADCommitId(folderName);
+            }
+
+            if (!snapshotInformation.ContainsKey(ProjectPathKey))
+            {
+                var repoRoot = GetRepoRoot(folderName);
+                var pathRelativeToRepoRoot = Path.GetDirectoryName(folderName)
+                    .Substring(repoRoot.Length)
+                    .Replace('\\', '/')
+                    .TrimStart('/');
+                snapshotInformation[ProjectPathKey] = pathRelativeToRepoRoot;
+            }
+        }
+
+        private string GetHEADCommitId(string folderName)
```

Comment:
Based on our naming convention, should be `GetHeadCommitId`. (We have `HttpSource` instead of `HTTPSource`)